### PR TITLE
Added dim parameter to tf.losses.softmax_cross_entropy

### LINKED ISSUE #20866: Why tf.losses.softmax_cross_entropy doesn't have "dim" (axis) argument?
Hi!
The function "tf.losses.softmax_cross_entropy" calls "nn.softmax_cross_entropy_with_logits_v2" which has a dim argument with a default value of -1.
Can we add the same "dim" or "axis" parameter to the calling function " tf.losses.softmax_cross_entropy", so that we can choose which axis to apply the softmax  ?
Thanks!

### PR DESCRIPTION
**Pull request by Siddhant085**
Added dim parameter to losses.softmax_cross_entropy. Fixes #20866

### CODE REVIEW

File: tensorflow/python/ops/losses/losses_impl.py [hunk 1]
```diff
@@ -739,7 +739,7 @@ def sigmoid_cross_entropy(
 
 @tf_export("losses.softmax_cross_entropy")
 def softmax_cross_entropy(
-    onehot_labels, logits, weights=1.0, label_smoothing=0, scope=None,
+    onehot_labels, logits, weights=1.0, label_smoothing=0, scope=None, dim=-1,
```
Comment (martinwicke) [inline tensorflow/python/ops/losses/losses_impl.py]: Should this be `axis`?

**Review by martinwicke — CHANGES_REQUESTED** [review]

**Comment by martinwicke** [comment]
(for API owners) Yes, should be axis. Fine otherwise.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 22 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by caisq** [comment]
@Siddhant085 Sorry for the delay, but please take a look at the test failures, e.g.,
https://source.cloud.google.com/results/invocations/f9de5bb5-3450-4f3c-b30a-e978b5e2203b/targets/%2F%2Ftensorflow%2Fpython%2Fkernel_tests:losses_test/log

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by tensorflowbutler** [comment]
It has been 43 days that this pull-request has stalled. Please create a new pull-request with the requested changes.
